crates/amaru/tests/evaluate_ledger_states.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2024 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use amaru_kernel::{
+    AnyCbor, AuxiliaryData, Bytes, Epoch, EraHistory, Hasher, KeepRaw, MintedTransactionBody, MintedTx,
+    MintedWitnessSet, Point, PoolId, TransactionInput, TransactionOutput, TransactionPointer, cbor, network::NetworkName, Network, Value,
+    protocol_parameters::ProtocolParameters,
+};
+use amaru_ledger::{
+    self,
+    context::DefaultValidationContext,
+    rules::transaction,
+    state::{self, volatile_db::StoreUpdate},
+    store::{EpochTransitionProgress, GovernanceActivity, ReadStore, Store, TransactionalContext},
+};
+use amaru_stores::rocksdb::RocksDB;
+use std::{collections::{BTreeMap, BTreeSet}, env, fs, iter, ops::Deref, path::PathBuf};
+
+use test_case::test_case;
+
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+
+struct TestContext {
+    ledger_dir: PathBuf,
+    pparams_dir: PathBuf,
+}
+
+fn get_test_context() -> Result<TestContext, Box<dyn std::error::Error>> {
+    let working_dir = env::current_dir()?;
+    Ok(TestContext {
+        ledger_dir: working_dir.join(PathBuf::from("../../ledger.db")),
+        pparams_dir: working_dir.join(PathBuf::from("../../cardano-blueprint/src/ledger/conformance-test-vectors/eras/conway/impl/dump/pparams-by-hash/")),
+    })
+}
+
+static TEST_MUTEX: Lazy<Mutex<()>> = Lazy::new(Mutex::default);
+
+include!("generated_ledger_conformance_test_cases.incl");
+
+fn evaluate_vector(snapshot: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let _shared = TEST_MUTEX.lock()?;
+    let network = NetworkName::Testnet(1);
+    let era_history = network.into();
+    match get_test_context() {
+        Ok(tc) => {
+            let vector_file = fs::read(snapshot)?;
+            let record: TestVector = minicbor::decode(&vector_file)?;
+            let () = import_vector(record, &tc.ledger_dir, era_history, &tc.pparams_dir)?;
+            Ok(())
+        }
+        Err(e) => {
+            tracing::warn!(
+                "skipping vector evaluation because some env vars are missing: {}",
+                e
+            );
+            Ok(())
+        }
+    }
+}
+
+fn peek_epoch(nes: &AnyCbor) -> Result<u64, Box<dyn std::error::Error>> {
+    let mut d = cbor::decode::Decoder::new(nes.raw_bytes());
+    let _ = d.array()?;
+    Ok(d.u64()?)
+}
+
+#[derive(cbor::Decode)]
+#[allow(dead_code)]
+struct TestVector {
```

Comment:
*Eventually*, though perhaps it's not needed right now, I'd love to get these `AnyCbor` types replaced with more strict types. Is there anything holding us back from that?